lib/vagrant-service-manager/binary_handlers/binary_handler.rb

```diff
@@ -100,10 +100,9 @@ def handle_windows_binary_path
           oc_version = CDKOpenshiftBinaryHandler::LATEST_OC_VERSION
         end
 
-        unless @options['--cli-version'] != oc_version || @options['--path']
-          path = PluginUtil.fetch_existing_oc_binary_path_in_windows
-          @path = path unless path.nil?
-        end
+        return if @options['--cli-version'] != oc_version || @options['--path']
```

Comment:
Right, you know that and it might work like this, but the code works heavily with side effects. If it completes the path is set, otherwise I take it that is exits earlier and somewhere else something is a check against _nil_ or so. I am not saying that it is not working. What I am saying is that it is not written in a clear way. Someone looking at it for the first time wonders about the intentions here.